ntoskrnl/mm/ARM3/virtual.c

```diff
@@ -5107,22 +5107,13 @@ NtFreeVirtualMemory(IN HANDLE ProcessHandle,
     BOOLEAN Attached = FALSE;
     PAGED_CODE();
 
-    //
-    // Only two flags are supported
-    //
-    if (!(FreeType & (MEM_RELEASE | MEM_DECOMMIT)))
-    {
-        DPRINT1("Invalid FreeType\n");
-        return STATUS_INVALID_PARAMETER_4;
-    }
-
-    //
-    // Check if no flag was used, or if both flags were used
-    //
-    if (!((FreeType & (MEM_DECOMMIT | MEM_RELEASE))) ||
-         ((FreeType & (MEM_DECOMMIT | MEM_RELEASE)) == (MEM_DECOMMIT | MEM_RELEASE)))
+    // Only two flags are supported, exclusively.
+    // NB: Check MEM_RELEASE first, as assumed to be used much more often.
+    if (FreeType != MEM_RELEASE && FreeType != MEM_DECOMMIT)
     {
-        DPRINT1("Invalid FreeType combination\n");
+        DPRINT1("Invalid FreeType (0x%08lx): "
+                 "only MEM_DECOMMIT and MEM_RELEASE are supported, exclusively\n",
```

Comment:
Printing the value is a good addition, but this explanation is not necessary in a debug print.

Reply:
Removed.